### PR TITLE
[Coral-Spark] Introduce SqlNode representation in CoralSpark

### PR DESCRIPTION
**Pull request by aastha25**
### What changes are proposed in this pull request, and why are they necessary?

This PR adds SqlNode as a class member for CoralSpark. This will enable downstream consumers to leverage the CoralSpark API to convert Coral IR to Spark compatible SqlNode representation and perform further transformations using SqlShuttles, as necessary. The final transformed SqlNode can be converted to SQL by utilizing `CoralSpark.constructSparkSQL(SqlNode)`.

### How was this patch tested?

./gradlew build

### CODE REVIEW

File: coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java [hunk 1]
```diff
@@ -241,4 +266,20 @@ public List<SparkUDFInfo> getSparkUDFInfoList() {
...
   }
+
+  /**
+   * Getter for the SqlNode representation
+   *
+   * @return SqlNode
+   */
+  public SqlNode getSqlNode() {
+    return sqlNode;
+  }
```
Comment (wmoustafa) [inline coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java:277]: I do not think we should "set" the SqlNode. Can we avoid that?

File: coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java [hunk 2]
```diff
@@ -241,4 +266,20 @@ public List<SparkUDFInfo> getSparkUDFInfoList() {
...
+
+  /**
+   * Setter for the Spark SQL
+   */
+  public void setSparkSql(String sparkSql) {
+    this.sparkSql = sparkSql;
```
Comment (wmoustafa) [inline coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java:283]: Passing Spark SQL in constructor as well as setter does not sound right. What is the reasoning? Can we avoid it?

Comment (aastha25) [reply coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java]: we dont need the setter with Spark-View-Catalog. Removed it

**Review by wmoustafa — APPROVED** [review]

File: coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java [hunk 3]
```diff
@@ -42,23 +43,26 @@ public class CoralSpark {
 
   private final List<String> baseTables;
   private final List<SparkUDFInfo> sparkUDFInfoList;
-  private final String sparkSql;
   private final HiveMetastoreClient hiveMetastoreClient;
+  private final SqlNode sqlNode;
```
Comment (ljfgem) [inline coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java:47]: I think `sparkSqlNode` is more accurate?

Comment (aastha25) [reply coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java:47]: Agreed, but let's introduce LanguageSqlNode terminology after the Coral IR APIs are completed and published.

**Pull request merged**